File: backend/app/schemas/project.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ProjectCreate(BaseModel):
    name: str
    description: str | None = None


class ProjectResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    name: str
    description: str | None
    created_at: datetime
    updated_at: datetime
    
class ProjectListResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    name: str
    description: str | None

    papers: int
    drafts: int
    progress: int
    status: str

    created_at: datetime
    updated_at: datetime

File: Client/paper-mind/src/types/project.ts
export interface Project {

    id: string;

    name: string;

    description: string;

    created_at: string;

    updated_at: string;

    papers?: number;

    drafts?: number;

    status?: string;

    progress?: number;
}

export interface CreateProjectInput {

    name: string;

    description: string;
}